edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/AuthChainHelpers.cs

```diff
@@ -12,11 +13,46 @@ public static string[] GetAuthChainIds(string authChain)
             return authChain.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
         }
 
+        public static Option<string> GetAuthTarget(Option<string> authChain)
+        {
+            if (!authChain.HasValue)
+            {
+                return Option.None<string>();
+            }
+
+            string authChainString = Preconditions.CheckNonWhiteSpace(authChain.OrDefault(), nameof(authChain));
+            string[] authChainIds = GetAuthChainIds(authChainString);
+
+            // The auth target is always the first element of the auth-chain
+            return authChainIds.FirstOption(id => true);
```

Comment:
auth chain goes like

edgeA;edgeB;device_z 
right? So the auth target here is edgeA?

Reply:
yep